dmscripts/data_retention_remove_supplier_declarations.py

```diff
@@ -8,3 +8,13 @@ def remove_supplier_data(data_api_client, logger, dry_run: bool):
         dry_run=dry_run
     )
     supplier_frameworks.remove_supplier_declaration_for_expired_frameworks()
+
+
+def remove_unsuccessful_supplier_declarations(data_api_client, logger, dry_run: bool, framework_slug: str):
+    supplier_frameworks_methods = SupplierFrameworkDeclarations(
+        api_client=data_api_client,
+        logger=logger,
+        dry_run=dry_run,
+        framework_slug=framework_slug
+    )
+    supplier_frameworks_methods.remove_declaration_from_failed_applicants()
```

Comment:
I'm not sure I understand what you're saying. Could you expand on what you're proposing?

Reply:
Sure 👍 
I guess my thinking is on L14 we're instantiating an object that is used to manipulate declarations. It's not (necessarily) an object that is used to manipulate only the declarations belonging to a single framework.

So passing the framework to the object rather than the method seems a bit restrictive.

A sort of metacode example from the DataApiClient would be:

```
class DataApiClient:
    def __init__(api_key):
        self.api_key = api_key
    def get_supplier(supplier_id):
        requests.get('api.marketplace.team/suppliers/{}'.format(supplier_id), auth=self.api_key)
```

Here we pass the `api_key` to the class on instantiation, because it relates to _every_ method and doesn't change in a single session. We pass the `supplier_id` to the `get_supplier` method because that method needs a supplier id.

So for SupplierFrameworkDeclarations it could look something like:

```
class SupplierFrameworkDeclarations:
    def __init__(api_client, logger, dry_run):
        self.api_client = api_client

    def remove_declaration_from_failed_applicants(framework_slug):
        suppliers = self.api_client.get_all_the_correct_suppliers_for_framework(framework_slug)
        for supplier in suppliers:
            self.api_client.remove_declaration(supplier.supplier_id, framework_slug)
```

And you'd call it like
```
supplier_frameworks_methods = SupplierFrameworkDeclarations(
        api_client=data_api_client,
        logger=logger,
        dry_run=dry_run,
    )
    supplier_frameworks_methods.remove_declaration_from_failed_applicants(framework_slug)
```

instead of:

```
supplier_frameworks_methods = SupplierFrameworkDeclarations(
        api_client=data_api_client,
        logger=logger,
        dry_run=dry_run,
        framework_slug=framework_slug
    )
    supplier_frameworks_methods.remove_declaration_from_failed_applicants()
```